src/args_parser.c

```diff
@@ -2,10 +2,22 @@
 
 #include <stdlib.h>
 
-args_t args_parse(char **argv) {
+args_t args_parse(int argc, char **argv) {
     args_t parsed;
+    parsed.flags[0].flag_name = "";
```

Comment:
This isn't necessary.

Reply:
I know that currently the function will not be called with argc less than two. but isn't it unsafe to have not initialized values being returned?

Reply:
Not really. These values are only used in comparison so even though they point to whatever it has in memory on that position it won't be a risk to us. Look at this piece of code and try to figure what are the values printed:

```
#include <stdio.h>
#include <string.h>

typedef struct sample_t {
	char *good;
} sample_t;

sample_t ret(char *s) {
	sample_t something;
	
	if (strlen(s) > 0) {
		something.good = s;
	}
	
	return something;
}

int main(void) {
	sample_t yes = ret("");
	sample_t no = ret("something");
	
	printf("%s\n", yes.good);
	printf("%s\n", no.good);
	
	return 0;
}
```